internal/http.go

```diff
@@ -0,0 +1,206 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	postgresURL    = "host=postgresql.test-project.svc  port=5432 user=curator password=M5rBgWkN8LfjeyI8 dbname=curatordb sslmode=disable"
+	httpServerPort = ":5000"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+type Response struct {
+	Message string `json:"message"`
+}
+
+func InitializeRouter() {
+	ctx, cancel := context.WithCancel(context.Background())
+	http.HandleFunc("/download", downloadReport)
+
+	httpServer := &http.Server{
+		Addr:        httpServerPort,
+		BaseContext: func(_ net.Listener) context.Context { return ctx },
+	}
+
+	reportLog.Info("starting http server on port 5000\n")
+	if err := httpServer.ListenAndServe(); err != nil {
+		log.Fatal(err)
+	}
+	httpServer.RegisterOnShutdown(cancel)
+
+	signalChan := make(chan os.Signal, 1)
+
+	signal.Notify(
+		signalChan,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGQUIT,
+	)
+
+	<-signalChan
+	reportLog.Info("os.Interrupt - shutting down...\n")
+
+	go func() {
+		<-signalChan
+		log.Fatal("os.Kill - terminating...\n")
+	}()
+
+	gracefullCtx, cancelShutdown := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancelShutdown()
+
+	if err := httpServer.Shutdown(gracefullCtx); err != nil {
+		reportLog.Info("http server shutdown error: %v\n", err)
+		defer os.Exit(1)
+	} else {
+		reportLog.Info("gracefully stopping http server\n")
+	}
+	defer os.Exit(0)
+}
+
+func downloadReport(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think we've reached the complexity threshold for this function and we'll need to work towards splitting up some of the logic into their own functions.